stackslib/src/chainstate/stacks/boot/mod.rs

```diff
@@ -1798,7 +1860,11 @@ pub mod test {
             boot_code_test_addr(),
             POX_4_NAME,
             "stack-extend",
-            vec![Value::UInt(lock_period), addr_tuple],
+            vec![
+                Value::UInt(lock_period),
+                addr_tuple,
+                Value::Optional(OptionalData { data: None }),
```

Comment:
The `make_pox_4_extend()` function needs to also take an `Option<StacksPublicKey>` so it can be added into this call to `stack-extend`.

Reply:
Generally speaking, you shouldn't put in default values to PoX-4 callers like `make_pox_4_extend`.  Anything you pass into `.pox-4` should be a Rust argument.